### PR TITLE
docs: update for attachFieldsToBody: keyValues

### LINKED ISSUE #489: Schema validation breaks when attachFieldsToBody is set to 'keyValues'
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the regression has not already been reported

### Last working version

7.7.3

### Stopped working in version

8.0.0

### Node.js version

18

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

13.6

### 💥 Regression Report

When setting `{attachFieldsToBody: 'keyValues'}`, the content of the file is attached to the body, using the name of the field.

According to the documentation here: https://github.com/fastify/fastify-multipart#json-schema-body-validation, we'd expect a string.

However, since version 8.0.0 (specifically, this particular change here: https://github.com/fastify/fastify-multipart/commit/c195d5f97d98d3b543b3d36fb82ffc03ec802646#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346L176, the object is attached as a Buffer.

Using JSON Schema, as far as I know, it is not possible to represent a Buffer object.

### Steps to Reproduce

- Install version 8.0.0
- Try to define a body schema that validates a buffer?

Here is a short example of a server:
```ts
import fastifyMultipart from "@fastify/multipart";
import fastify from "fastify";

const app = fastify({ ignoreTrailingSlash: true });
app.register(fastifyMultipart, {
  attachFieldsToBody: "keyValues",
});

app.post(
  "/test",
  {
    schema: {
      body: {
        type: "object",
        properties: {
          myFile: { type: "string" },
        },
      },
    },
  },
  async (request, reply) => {
    reply.send();
  }
);

app.listen({ port: 3000 });
```

It works fine in 7.7.3. With 8.0.0, `myFile` is now a Buffer, and this server returns this:

```json
{
    "statusCode": 400,
    "error": "Bad Request",
    "message": "body/myFile must be string"
}
```

### Expected Behavior

- `body.myFile` should be a string, or there is a documented clear way of defining a Buffer in the JSON Schema validation

### PR DESCRIPTION
**Pull request by dancastillo**
#### Checklist

closes: #489 

- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -288,7 +288,7 @@ If you try to read from a stream and pipe to a new file, you will obtain an empt
 
 ## JSON Schema body validation
 
-If you enable `attachFieldsToBody: 'keyValues'` then the response body and JSON Schema validation will behave similarly to `application/json` and [`application/x-www-form-urlencoded`](https://github.com/fastify/fastify-formbody) content types. Files will be decoded using `Buffer.toString()` and attached as a body value.
+If you enable `attachFieldsToBody: 'keyValues'` then the response body and JSON Schema validation will behave similarly to `application/json` and [`application/x-www-form-urlencoded`](https://github.com/fastify/fastify-formbody) content types. Files will be decoded using `Buffer` and attached as a body value.
```
Comment (gurgunday) [inline README.md:291]: I know this whole README is full of "you"s and all, but one commit at a time, we need to do better 🤣

```suggestion
When the `attachFieldsToBody` parameter is set to `'keyValues'`, the response body and JSON Schema validation will behave in a manner similar to that observed with the `application/json` and [`application/x-www-form-urlencoded`](https://github.com/fastify/fastify-formbody) content types. Additionally, files will be attached to the body as Buffer objects.
```

Comment (gurgunday) [reply README.md:291]: Ref: https://fastify.dev/docs/latest/Guides/Style-Guide

Comment (dancastillo) [reply README.md]: Good call out. Just updated

**Review by gurgunday — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (gurgunday) [reply README.md]: Thanks! 🚀

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**